OPHD/StructureManager.h

```diff
@@ -37,6 +37,38 @@ class SKey
 };
 
 
+/**
+ * Object suitable for range-based for loops.
+ * This class allows code to iterate over instances of a given component
+ * without exposing the internal component storage container type.
+ */
+template<typename ComponentTy>
+class ComponentRange
+{
+private:
+	class Iterator
+	{
+	private:
+		std::map<SKey, StructureComponent*>::iterator mIt;
+	public:
+		Iterator(std::map<SKey, StructureComponent*>::iterator it) : mIt(it) {}
+		bool operator!= (const Iterator& rhs) const { return mIt != rhs.mIt; }
+		Iterator& operator++() { ++mIt; return *this; }
+		operator ComponentTy* () const { return static_cast<ComponentTy*>(mIt->second); }
+		ComponentTy* operator->() const { return static_cast<ComponentTy*>(mIt->second); }
```

Comment:
The spacing is a bit inconsistent here before the opening parenthesis for the parameter list. I would recommend no spacing.